app/Repositories/ModularPageRepository.php

```diff
@@ -0,0 +1,149 @@
+                if (str_contains($component, 'modular')) {
+                    // Modify field name to match component filename
+                    $component = str_replace('modular-', '', $component);
+                    $component = str_replace('_', '-', $component);
+
+                    if (str_starts_with($properties, '{') === true) {
+                        $components[$component] = $this->parsePromoJSON($properties, $data['page']['id']);
+                    } else {
+                        // If only an ID is entered without json
+                        $components[$component]['id'] = (int)$properties;
+                        $components[$component]['config'] = '';
+                    }
+
+                    // Parse promos
+                    $group_reference[$components[$component]['id']] = $component;
+                    $group_config[$component] = $components[$component]['config'];
+                }
+            }
+        }
+
+        $params = [
+            'method' => 'cms.promotions.listing',
+            'promo_group_id' => array_keys($group_reference),
+            'filename_url' => true,
+            'is_active' => '1',
+        ];
+
+        $promos = $this->cache->remember($params['method'].md5(serialize($params)), config('cache.ttl'), function () use ($params) {
+            return $this->wsuApi->sendRequest($params['method'], $params);
+        });
+
+        $promos = $this->appendComponentProperties($promos, $components);
+
+        $promos = $this->parsePromos->parse($promos, $group_reference, $group_config);
+
+        // Reset promo item key to use component values in template
+        if (!empty($promos)) {
+            foreach ($promos as $component => $values) {
+                $promos[$component] = array_values($promos[$component]);
+            }
+        }
+
+        return $promos;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function parsePromoJSON($json, $page_id)
+    {
+        $component = [];
+
+        // Remove all spaces and line breaks
+        $component = preg_replace('/\s*\R\s*/', '', $json);
+
+        // Last item cannot have comma at the end of it
+        $component = preg_replace('(,})', '}', $component);
+
+        // JSON to array
+        $component = json_decode($component, true);
+
+        // Make sure config always exists
+        $component['config'] = (!empty($component['config']) ? $component['config'] : '');
+
+        // Append actual page id to config
+        if (str_contains($component['config'], 'page_id')) {
+            $component['config'] = preg_replace('/\bpage_id\b/', 'page_id:'.$page_id, $component['config']);
+        }
+
+        // Not allowing "first" config option
+        // API seems to do fine with double pipes, so not handling them at this time
+        if (str_contains($component['config'], 'first')) {
+            $component['config'] = preg_replace('/\bfirst\b/', '', $component['config']);
```

Comment:
This feels safe to do without the conditional around it. Is there a config situation where config options would have the chars `first` in it that are not prefixed or suffixed by a pipe? 
https://github.com/waynestate/parse-promos/blob/master/README.md#usage

Reply:
@nickdenardis The issue is which pipe to remove, we can't be sure someone won't put it at the beginning or end of the config. Example `randomize|page_id|first` is valid, or where the only config option is `first` without any pipes. If anyone were to accidentally use first, it would break the page, so I have to be sure its removed.